hack/generate-kustomize-patch-gardenlet.sh

```diff
@@ -5,6 +5,24 @@
 set -e
 set -o pipefail
 
+# Try to acquire a lockfile to write the imagevector override patch as
+# else skaffold's concurrent execution can cause conflicting writes.
+LOCKFILE="/tmp/generate-kustomize-patch-gardenlet.sh.lock"
+LOCKFILE_DEST=""
+# Try to acquire the lock in a loop
+while ! ln -s "$$" "$LOCKFILE" 2>/dev/null; do
+  OLD_LOCKFILE_DEST=$LOCKFILE_DEST
+  LOCKFILE_DEST=$(readlink $LOCKFILE)
+  if [[ $OLD_LOCKFILE_DEST == $LOCKFILE_DEST ]]; then
+    echo "Lock file ($LOCKFILE) has not changed recently. Proceeding..."
+    break
+  fi
+  echo "Another instance ($LOCKFILE_DEST, lock file $LOCKFILE) is running. Waiting..."
+  sleep 1  # Wait before retrying
+done
+# Ensure lock is removed on exit
+trap 'rm -f "$LOCKFILE"' EXIT
```

Comment:
Can you move this into a function and make it reusable for the other `hack/generate-kustomize-*.sh` scripts as well? They are only called once/not in parallel as of today, but who knows whether it will stay like this. We don't wanna run into the same flake in the future :)